[clang]MveEmitter:Pass Args as const references *bad rebase, please ignore*

aniplcc opened the pull request:
Closes  #89192

vfdff committed:
[InstCombine] Add check to avoid dependent optimization order, NFC

Since PR86428, foldPowiReassoc is called by both FMul and FDiv,
as the optimization of FDiv is placed after the FMul, so now
it is correct we don't add the checking of FDiv for powi(X, Y) / X.
But, we may add more matching scenarios later, so add the checking opcode
explicitly is easier to understand.

vfdff committed:
[InstCombine] Optimize powi(X, Y)/ (X * Z) with Ofast

foldFDivPowDivisor can address A / powi(x, y) to A * powi(x, -y),
while for small const value y, for example y=2, the instcombine will
transform powi(x, 2) to fmul x, x, so it is not optimal for A / powi(x, 2).

Fix https://github.com/llvm/llvm-project/issues/77171

tbaederr committed:
[clang][Interp] Support __builtin_os_log_format_buffer_size

aniplcc committed:
[clang]MveEmitter:Update args with const references

alexey-bataev committed:
[SLP]Fix final analysis for unsigned nodes.

Need to check that at least single bit is cleared for unsigned nodes
before reducing their size. Otherwise they might be treated as signed in
signed nodes.

VyacheslavLevytskyy committed:
[SPIR-V] SPIR-V Backend must generate a valid OCL version if working in OpenCL environment (#89199)

If there is no information about OpenCL version we are forced to
generate OpenCL 1.0 by default for the OpenCL environment to avoid
puzzling run-times with Unknown/0.0 version output. For a reference,
LLVM-SPIRV Translator avoids potential issues with run-times in a
similar manner.

martinboehme committed:
[clang][dataflow][NFC] Fix code formatting in DataflowEnvironment.cpp (#89352)

For some reason, when I merged #89235, two lines were mis-formatted.

This patch corrects this; while I'm here, I'm also correcting other
existing formatting errors.

ChuanqiXu9 committed:
[NFC] [Serialization] Use semantical type 'DeclID' for 'CreateDeserialized'

Previously we use 'unsigned' as the type of ID in 'CreateDeserialized'.

And the type of `DeclID` in serialization is 'uint32_t', so there is
minor inconsistency.

Also more importantly, if we want to extend the type of DeclID from
uint32_t to uint64_t, we may be in trouble due to we forgot updating the
a lot of 'CreateDeserialized'.

So this patch tries to use semantical type 'DeclID' for
'*Decl::CreateDeserialized' to make sure it is tightly consistent.

davemgreen committed:
[AArch64] Remove invalid uabdl patterns. (#89272)

These were added in https://reviews.llvm.org/D14208, which look like
they attempt to detect abs from xor+add+ashr. They do not appear to be
detecting the correct value for the src input though, which I think is
intended to be the sub(zext, zext) part of the pattern. We have pattens
from abs now, so the old invalid patterns can be removed.

Fixes #88784

fhahn committed:
[VPlan] Introduce recipes for VP loads and stores. (#87816)

Introduce new subclasses of VPWidenMemoryRecipe for VP
(vector-predicated) loads and stores to address multiple TODOs from
https://github.com/llvm/llvm-project/pull/76172

Note that the introduction of the new recipes also improves code-gen for
VP gather/scatters by removing the redundant header mask. With the new
approach, it is not sufficient to look at users of the widened canonical
IV to find all uses of the header mask.

In some cases, a widened IV is used instead of separately widening the
canonical IV. To handle that, first collect all VPValues representing header
masks (by looking at users of both the canonical IV and widened inductions
that are canonical) and then checking all users (recursively) of those header
masks.

Depends on https://github.com/llvm/llvm-project/pull/87411.

PR: https://github.com/llvm/llvm-project/pull/87816

TinaAMD committed:
[mlir][emitc] Restrict types in EmitC (#88391)

Restrict the types which are valid for EmitC operations. Use what is
currently supported by the emitter as restriction. Define a utility
functions for valid types, such that they can be used to restrict the
operations in table gen as well as being available for reuse in dialect
conversions.

Pierre-vh committed:
[AMDGPU] Allow any linkage for dynlds (#84742)

Solves SWDEV-449592

artagnon committed:
MachineInstrBundle: modernize a couple of loops (NFC) (#89229)

dtcxzyw committed:
[IR] Add support for trunc's nuw/nsw flags in `copyIRFlags` (#89353)

This patch fixes
https://github.com/llvm/llvm-project/pull/85592#issuecomment-2065865020.

I found this while fixing flag propagation in my
["vectorizer"](https://github.com/dtcxzyw/llvm-codegen-benchmark/blob/main/vectorize.cpp).

metaflow committed:
Revert "[SLP]Fix final analysis for unsigned nodes."

This reverts commit 74e07ab523122d6a8347b25770062ab331b6bb84.

It might be that Mask.getBitWidth() == Mask.countl_zero() (32 in my
case) and zero bitwidth2 causes the crash.

Commit:
[AArch64][SVE2] Generate SVE2 BSL instruction in LLVM for add/sub. (#88413)

Allow to fold or/and-and to BSL instuction for scalable vectors.

RKSimon committed:
[AArch64][X86] xor.ll - fix vec_add_of_not_with_undef_decrement copy+pasta typo

This was copied from vec_add_of_not_with_undef instead of vec_add_of_not_decrement - replacing the second sub with an add

RKSimon committed:
[DAG] visitADDLike - update "(x - y) + -1  ->  add (xor y, -1), x" fold to accept UNDEF in a splat vector of -1

Make sure we use getNOT instead of reusing the allones (with undefs) vector

Dinistro committed:
[MLIR][DataLayout] Add support for scalable vectors (#89349)

This commit extends the data layout to support scalable vectors. For
scalable vectors, the `TypeSize`'s scalable field is set accordingly,
and the alignment information remains the same as for normal vectors.
This behavior is in sync with what LLVM's data layout queries are
producing.

Before this change, scalable vectors incorrectly returned the same size
as "normal" vectors.

kiranchandramohan committed:
[Flang][OpenMP] Issue error if reduction clause has proc-pointer (#88999)

OpenMP 5.2: Section 5.5.5: A procedure pointer must not appear in a
reduction clause.

Fixes #87915

Dinistro committed:
[MLIR][LLVM] Add flag to skip import of DICompositeType's elems (#89355)

This commit introduces a flag to allow skipping the potentially
recursive import of DICompositeType elements. This patch is essentially a
bandaid for the still broken recursive debug type import.

Some of our downstream inputs are produced by excessive usage of
template meta programming, and thus contain tens of thousands of types
that all participate in such recursions. Unfortunately, the series of
patches that introduces type support is not easily revertible due to
being around for a while now and Modular depending on it.

We can consider to revert this change once the type importer has show to
be very performant, but for now we are talking second vs hours to import
specific files.

redstar committed:
[SystemZ][z/OS] Implement llvm.frameaddr for XPLINK (#89284)

The implementation follows the ELF implementation.

labath committed:
[lldb] Make SBType::FindDirectNestedType work with expression ASTs (#89183)

The types we get out of expressions will not have an associated symbol
file, so the current method of looking up the type will fail. Instead, I
plumb the query through the TypeSystem class. This correctly finds the
type in both cases (importing it into the expression AST if needed). I
haven't measured, but it should also be more efficient than doing a type
lookup (at least, after the type has already been found once).

dominik-steenken committed:
[SystemZ] Fix Operand Retrieval for Vector Reduction Intrinsic in `shouldExpandReduction` (#88874)

In the existing version, SystemZTTIImpl::shouldExpandReduction will
create a `cast` error when handling vector reduction intrinsics that
do not have the vector to reduce as their first operand, such as
`llvm.vector.reduce.fadd` and `llvm.vector.reduce.fmul`.
This commit fixes that problem by moving the cast into the case
statement that handles the specific intrinsic, where the vector
operand position is well-known.

NagyDonat committed:
[analyzer] Use explicit call description mode (easy cases) (#88879)

This commit explicitly specifies the matching mode (C library function,
any non-method function, or C++ method) for the `CallDescription`s
constructed in various checkers where this transition was easy and
straightforward.

This change won't cause major functional changes, but isn't NFC because
it ensures that e.g. call descriptions for a non-method function won't
accidentally match a method that has the same name.

Separate commits will perform (or have already performed) this change in
other checkers. My goal is to ensure that the call description mode is
always explicitly specified and eliminate (or strongly restrict) the
vague "may be either a method or a simple function" mode that's the
current default.

Commit:
[AIX][Debug]correct the cases on AIX bot, NFC

Related to commit b2323f43e3cdb52b4e15a7d4f434cd5c64740dd4

jayfoad committed:
[KnownBits] Simplify optimality checking in unit tests. NFC. (#89368)

Use bool instead of function_ref.

dominik-steenken committed:
[SystemZ] Add TPEI instruction and Associated Facility (#89372)

This PR adds the TPEI (Test Pending External Interruption) instruction,
along with the facility that contains it. This is a millicoded system
instruction that is not used for code generation, so it will be used
exclusively by the Assembler and Disassembler.
Accordingly, this commit also adds tests for both.

xingxue-ibm committed:
[OpenMP] Use half of available logical processors for collapse tests (#88319)

The new collapse test cases define `MAX_THREADS` to be 256 and use all
available threads/logical processors on the system. This triples the
testing time on an AIX machine that has 128 logical processors. This
patch changes to use half of available logical processors to avoid over
subscribing because there are other libomp tests running at the same
time, including 2 other such collapse tests.

arsenm committed:
AMDGPU: Refactor unsafe atomicrmw remark emission (#89379)

Defers some of the setup work until the remark is actually emitted. Move
big lambda into a helper function, and only use value capture of a
pointer. Prepare to have different messages.

arsenm committed:
AMDGPU: Use common check prefix in atomic expand test

kadircet committed:
[clangd] Fix shared-lib builds after 2cdbc9cff3b7ef262

chsigg committed:
Switch member calls to `isa/dyn_cast/cast/...` to free function calls. (#89356)

This change cleans up call sites. Next step is to mark the member
functions deprecated.

See https://mlir.llvm.org/deprecation and
https://discourse.llvm.org/t/preferred-casting-style-going-forward.

Commit:
[Sema] Check if types are resolved before querying function description.

jayfoad committed:
Add IIT_V10 to support 10-element vectors in intrinsics (#89383)

Needed for a future patch.

steakhal committed:
[analyzer] Fix stores through label locations (#89265)

Interestingly, this case crashed from the very beginning of the project,
at least starting by clang-3.

As a "fix" I just do the same thing as we do for concrete integers. It
might not be the best we could do, but arguably, it's still better than
crashing.

Fixes #89185

clementval committed:
[flang][cuda] Allow if stmt in device subroutine (#89347)

pogo59 committed:
[NFC][X86][Headers] Fix missing blank line

It caused the \headerfile directive to be considered as part of
the brief description in some of our tooling.

yxsamliu committed:
[ClangOffloadBundler] Add file size to header (#88827)

__hipRegisterFatBinary only accepts one pointer argument. It is expected
to get the fat binary size from the header.

This patch adds a file size field to the header of the compressed
bundle.

adam-smnk committed:
[mlir][xegpu] XeGPU alias ops folder pass (#88886)

Adds a pass that folds aliasing ops into XeGPU ops.

Commit:
Reapply "[Clang][AArch64] Warn when calling non/streaming about vector size difference (#79842)"

This reverts commit 950bb097e11d6ee26533c00519c62df994322228

KoolJBlack committed:
[mlir][ArmNeon] Update `LowerContractionToSMMLAPattern` to support proper unrolling for k dimension (#88591)

Fixes correctness issue with current smmla unrolling patterns whereby
unrolling K dimension would only include the result from the last tile
along K. Updates patterns to feed previous smmla output of the previous
tile into the next one along K.

ZelinMa557 committed:
[InstCombine] Regard zext nneg as sext when folding add(zext neg(add)) (#88887)

fixes #88348
proof:
https://alive2.llvm.org/ce/z/fJnM7t
test will be added later

---------

Signed-off-by: ZelinMa557 <3388706467@qq.com>

Dinistro committed:
[MLIR][LLVM] Add vector exception to composite type element ignore mode (#89385)

This commit fixes a bug in the DICompositeType element ignore mode. It
seems that vectors require the presence of elements, as they otherwise
do not pass the verifier.

skatrak committed:
[Flang][OpenMP] NFC: Simplify handling of insertion points (#89221)

This patch replaces some `saveInsertionPoint`, `restoreInsertionPoint`
call pairs for an `InsertionGuard` instance where it makes sense within
Flang OpenMP lowering to make further modifications less error-prone.

skatrak committed:
[MLIR][OpenMP] Verify loop wrapper properties of omp.parallel (#88722)

This patch extends verification of the `omp.parallel` operation to check
it is correctly defined when taking a loop wrapper role.

In OpenMP, a PARALLEL construct can be either a (potenially combined)
block construct or a loop construct, when appearing as part of a
composite construct. This is currently the case for the DISTRIBUTE
PARALLEL DO/FOR and DISTRIBUTE PARALLEL DO/FOR SIMD exclusively.

When used to represent the PARALLEL leaf of a composite construct, it
must follow the rules of a wrapper loop operation in MLIR, and this is
what this patch ensures. No additional restrictions are introduced for
PARALLEL block constructs.

tbaederr committed:
[clang][Interp] Change array index types in OffsetHelper

This is closer to that the current interpreter does. It also fixes
diagnostics in a case I was looking into. Unfortunately, this is not
possible to test right now since it requires a large array and we
don't implement array fillers yet.

cor3ntin committed:
[Clang] Fix the mangling of lambdas (#89204)

Lambdas used in the initializer of a local class were not mangling the
name of the member.

Fixes #88906

vzakhari committed:
Reland "[flang][build] Fixed paths discrovery for the out-of-tree build. (#87822)"

This reverts commit 215eee60497489ae0cc7cc78c0d8b8270e057a70.

RKSimon committed:
[TableGen] MacroFusionPredicatorEmitter - pass constant std::vector arguments by ArrayRef instead

Silence pass by value warnings

Fixes #89210

MaskRay committed:
[ELF] Add --default-script/-dT

GNU ld added --default-script (alias: -dT) in 2007. The option specifies
a default script that is processed if --script/-T is not specified. -dT
can be used to override GNU ld's internal linker script, but only when
the application does not specify -T.
In addition, dynamorio's CMakeLists.txt may use -dT.

The implementation is simple and the feature can be useful to dabble
with different section layouts.

Pull Request: https://github.com/llvm/llvm-project/pull/89327

nickdesaulniers committed:
[libc] fixup nascent pthread_condattr_test (#89308)

- use namespaced identifiers
- add corresponding headers for namespaced declarations
- replace time.h and errno.h with finer grain includes
- update cmake

Fixes: #88987
Fixes: #89261
Link: #88997
Link: #89262

MaskRay committed:
[llvm-readelf] Print more information for RELR

llvm-readelf/llvm-readobj print RELR as REL relocations with a fixed
type (e.g. `R_*_RELATIVE`). GNU readelf printed only addresses and have
recently switched to a more descritive style that includes a symbolic
address column (symbolized using .symtab/.strtab) (milestone: binutils
2.43).

This patch implements the new GNU style, which seems superior to the
current REL style and essentially obsoletes LLVM-specific --raw-relr
(`printRelrReloc`).

Pull Request: https://github.com/llvm/llvm-project/pull/89162

Commit:
[mlir][llvm] Port `overflowFlags` to a native operation property (RELAND) (#89410)

This PR changes the LLVM dialect's IntegerOverflowFlags to be stored on
operations as native properties.

Reland to fix flang

aeubanks committed:
[clang] Add flag to experiment with cold function attributes (#89298)

To be removed and promoted to a proper driver flag if experiments turn
out fruitful.

For now, this can be experimented with `-mllvm
-pgo-cold-func-opt=[optsize|minsize|optnone|default] -mllvm
-enable-pgo-force-function-attrs`.

Original LLVM patch for this functionality: #69030

Troy-Butler committed:
[LLVM][TextAPI] Refactor Redundant Condition (#86542)

Fixes #83241

---------

Signed-off-by: Troy-Butler <squintik@outlook.com>
Co-authored-by: Troy-Butler <squintik@outlook.com>

yijia-g committed:
[mlir][bazel] add bazel rule for XeGPU

slackito committed:
[bazel][mlir] Fix invalid quotation mark in BUILD file

MaskRay committed:
[ELF,test] Remove --raw-relr from pack-dyn-relocs.s

RKSimon committed:
[X86] LowerFunnelShift - improve handling of vXi8 constant splat funnel shifts

This patch moves the promotion to vXi16 shifts and the upper/lower bit masking into LowerFunnelShift for targets that have a bit-select instruction (XOP's VPCMOV and AVX512's VPTERNLOG).

This prevents the regressions in #89115 due to the masking of ((X << V) | (Y >> (8-V))) vXi8 shifts.

LRFLEW committed:
linear_congruential_engine: add using more precision to prevent overflow (#81583)

This PR is a followup to #81080.

This PR makes two major changes to how the LCG operation is computed:

The first is that I added an additional case where `ax + c` might
overflow the intermediate variable, but `ax` by itself won't. In this
case, it's much better to use `(ax mod m) + c mod m` than the previous
behavior of falling back to Schrage's algorithm. The addition modulo is
done in the same way as when using Schrage's algorithm (i.e. `x += c -
(x >= m - c)*m`), but the multiplication modulo is calculated directly,
which is faster.

The second is that I added handling for the case where the `ax`
intermediate might overflow, but Schrage's algorithm doesn't apply (i.e.
r > q). In this case, the only real option is to increase the precision
of the intermediate values. The good news is that - for `x`, `a`, and
`c` being n-bit values - `ax + c` will never overflow a 2n-bit
intermediary, meaning this promotion can only happen once, and will
always be able to use the simplest implementation. This is already the
case for 16-bit LCGs, as libcxx chooses to compute them with 32-bit
intermediate values. For 32-bit LCGs, I simply added code similar to the
16-bit case to use the existing 64-bit implementations. Lastly, for
64-bit LCGs, I wrote a case that calculates it using `unsigned __int128`
if it is available to use.

While this implementation covers a *lot* of the missing cases from
#81080, this still won't compile **every** possible
`linear_congruential_engine`. Specifically, if `a`, `c`, and `m` are
chosen such that it needs 128-bit integers, but the platform doesn't
support `__int128` (eg. 32-bit x86), then it will fail to compile.
However, this is a fairly rare case to see actually used, and libcxx
would be in good company with this, as [libstdc++ also fails to compile
under these
circumstances](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=87744).
Fixing **this** gap would require even **more** work of further
complexity, so that would probably be best handled by a different PR
(I'll put more details on what that PR would entail in a comment).

MatzeB committed:
GlobalsModRef, ValueTracking: Look through threadlocal.address intrinsic (#88418)

This improves handling of `threadlocal.address` intrinsic in analyses:

The thread-id cannot change within a function with the exception of
suspend points of pre-split coroutines. This changes
`llvm::getUnderlyingObject` to look through `threadlocal.address` in
these cases.

`GlobalsAAResult::AnalyzeUsesOfPointer` checks whether an address can be
traced to simple loads/stores or escapes to other places. Starting the
analysis from a thread-local `GlobalValue` the `threadlocal.address`
intrinsic is safe to skip here.

This improves issue #87437

sdkrystian committed:
[libc++] Fix usage of 'exclude_from_explicit_instantiation' attribute on local class members (#89377)

#88777 adds a warning for when the `exclude_from_explicit_instantiation`
attribute is applied to local classes and members thereof. This patch
addresses the few instances of `exclude_from_explicit_instantiation`
being applied to local class members in libc++.

rsuderman committed:
[mlir][tensor] Fold `tensor.reshape` for dynamic reshape (#88961)

If `tensor.reshape` occurs with `d0, d1, d2, ...` for the dimensions we
know that the reshape is a no-op. Checking for this case lets us fold
away the computation.

mjklemm committed:
[flang] Remove obsolete flang-to-external-fc tool (#88904)

It seems like the `flang-to-external-fc` tool is no longer needed,
because Flang is now a full compiler in its own right. After PR #85249
has landed, this tool will not be able to pick up the `.f18.mod` files.

darkbuck committed:
[test] Fix test. NFC

- Add the missing token in @llvm.coro.end

MaskRay committed:
[Driver] Add m_Group to -m[no-]strict-align

so that `Args.ClaimAllArgs(options::OPT_CompileOnly_Group);` in
Driver::handleArguments will claim the options even if only linking is
performed.

Fix #87945

fabio-d committed:
Sync FuchsiaConfig with downstream's custom_scudo_config.h (#89244)

Downstream disabled EnableContiguousRegions on RISCV-64 to avoid
running out of virtual memory, but our tests still use the internal
FuchsiaConfig class, which therefore needs to be changed too.

alexey-bataev committed:
[SLP]Fix PR89438: Check for same vectorized node in MinBWs, not user.

Need to check if the buildvector node has perfect diamond match in the
graph and the matched node is resized.

topperc committed:
[SelectionDAG] Require UADDO_CARRY carryin and carryout to have the same type. (#89255)

This requires type legalization to keep them the same. This means we no
longer need to legalize the operand since it will be legalized when we
legalize the second result.

topperc committed:
[RISCV] Rename FeatureRVE to FeatureStdExtE. NFC (#89174)

Planning to declare all extensions in tablegen so we can generate the
tables for RISCVISAInfo.cpp. This requires making "e" consistent with
other extensions.

topperc committed:
[X86][TableGen] Remove unnecessary use of formatted_raw_ostream. NFC (#89343)

This code used to use the PadToColumn feature of formatted_raw_ostream,
but no longer does. formatted_raw_ostream is slower than regular
raw_ostream because it has to keep track of the number of character
since the last new line character.

jansvoboda11 committed:
[clang][modules] Only avoid pruning module maps when asked to (#89428)

Pruning non-affecting module maps is useful even when passing module
maps explicitly via `-fmodule-map-file=<path>`. For this situation, this
patch reinstates the behavior we had prior to #87849. For the situation
where the explicit module map file arguments were generated by the
dependency scanner (which already pruned the non-affecting ones), this
patch introduces new `-cc1` flag
`-fno-modules-prune-non-affecting-module-map-files` that avoids the
extra work.

clementval committed:
[flang][cuda] Allow fixed size array with the managed attribute (#89436)

Fixed size array and scalar should be allowed with the `managed`
attribute.

bwendling committed:
[Clang] Handle structs with inner structs and no fields (#89126)

A struct that declares an inner struct, but no fields, won't have a
field count. So getting the offset of the inner struct fails. This
happens in both C and C++:

  struct foo {
    struct bar {
      int Quantizermatrix[];
    };
  };

Here 'struct foo' has no fields.

Closes: https://github.com/llvm/llvm-project/issues/88931

jsjodin committed:
[flang][OpenMP] Use maxnum/minnum for lowering of max/min reduction operators (#89258)

This patch changes the lowering of max and min to be lowered to
arith::MaxNumFop and arith::MinNumFOp instead of using arith::MaximumFOp
and arith::MinimumFOp. The arith::MaximumFOp and arith::MinimumFOp map
to the corresponding intrinsics llvm.maximum.* and llvm.minimum.*
intrinsics which conform to the semantics specified in the draft of IEEE
754-2019, which is not supported by all hardware. Instead using
arith::MaximumFOp and arith::MinimumFOp will allow code generation for
more targets and match the code generated by clang OpenMP.

fixes #87955

hokein committed:
[clang] CTAD: Fix require-clause is not transformed. (#89378)

Fixes https://github.com/llvm/llvm-project/issues/89013

When building the deduction guide, we use the
TemplateArgsForBuildingFPrime to transform the require-clause from the
underlying class deduction guide. However, we do this at the wrong place
where not all elements of TemplateArgsForBuildingFPrime are initialized.
The fix involves rearranging the transformRequireClause call to the
correct location.

As part of the fix, we extend the TemplateInstantiator to support more
types in the template-rewrite mode. Otherwise, we will encounter an
assertion error when attempting to rewrite the template type parameter
type like D with a complex type like Derived<U>.

adrian-prantl committed:
[lldb] Provide a better error message for missing symbols (#89433)

This adds a hint to the missing symbols error message to make it easier
to understand what this means to users.

adrian-prantl committed:
[lldb] Provide a better error message for missing symbols (#89433)

This adds a hint to the missing symbols error message to make it easier
to understand what this means to users.

[Reapplies an earlier patch with a test fix.]

bazuzi committed:
[clang][dataflow] Expose getReferencedDecls for a Stmt. (#89444)

nicovank committed:
Apply modernize-use-starts-ends-with on llvm-project (#89140)

Run `modernize-use-starts-ends-with` on llvm-project. Two instances are
flagged, minor readability improvements, extremely minor performance
improvements.

```
python3 clang-tools-extra/clang-tidy/tool/run-clang-tidy.py \
    -clang-tidy-binary="build/bin/clang-tidy" \
    -clang-apply-replacements-binary="build/bin/clang-apply-replacements" \
    -checks="-*,modernize-use-starts-ends-with" \
    -header-filter=".*" \
    -fix -format
```

I am working on some additions to this check, but they don't seem to
flag any additional cases anyway.

efriedma-quic committed:
[ARM64EC] Add softintrin.lib as an implicit dependency to object files. (#89171)

This copies MSVC behavior, and avoids weird link errors in certain
cases.

bwendling committed:
[Clang] Loop over FieldDecls instead of all Decls (#89453)

Only FieldDecls are of importance here. A struct defined within another
struct has the same semantics as if it were defined outside of the
struct. So there's no need to look into RecordDecls that aren't a field.

clementval committed:
[flang][cuda] Use fir.cuda_deallocate for automatic deallocation (#89450)

Automatic deallocation of allocatable that are cuda device variable must
use the fir.cuda_deallocate operation. This patch update the automatic
deallocation code generation to use this operation when the variable is
a cuda variable.

makslevental committed:
[mlir][python] fix memref._is_constant_int_like (#89447)

clementval committed:
[flang][cuda] Enforce PINNED attribute when ALLOCATE with PINNED option (#89455)

When the PINNED option is specified on an ALLOCATE statement, the object
must have the PINNED attribute.

clementval committed:
[flang][cuda] Enforce DEVICE attribute when ALLOCATE with STREAM option (#89459)

When the STREAM option is specified on an ALLOCATE statement, the object
must have the DEVICE attribute.

MaskRay committed:
[BOLT,test] Update AArch64/constant_island_pie_update.s after llvm-readelf -r RELR change

arsenm committed:
AMDGPU: Add tests for atomicrmw handling of new metadata (#89248)

Add baseline tests which should comprehensively test the new atomic
metadata. Test codegen / expansion, and preservation in a few
transforms.

New metadata defined in #85052

clementval committed:
[flang][cuda] Relax assumed size check on object with device attribute (#89466)

Assumed size arrays are apparently allowed with attribute device.

topperc committed:
[RISCV] Add RISCVTuneProcessorModel to 'generic' CPU. NFC

Remove hardcode GENERIC cpu from RISCVTargetDefEmitter.cpp.

vitalybuka committed:
[test][sancov] Regenerate with update_test_checks (#89457)

Prepare for #89458

adrian-prantl committed:
Revert "[lldb] Provide a better error message for missing symbols (#89433)"

This reverts commit 08163cd9d82690e808c28515523b5fd0923d7b38.

I accidentally broke the test while addressing review feedback.

HerrCai0907 committed:
[clang-tidy] bugprone-lambda-function-name ignore macro in captures (#89076)

kazutakahirata committed:
[memprof] Drop the trait parameter (NFC) (#89461)

OnDiskIterableChainedHashTable::Create can default-contruct a trait
object for us.  We don't need to construct one on our own unless we
need to customize something (like a version number).

vitalybuka committed:
[sancov] Apply branch weights when checking counters (#89458)

It reduces instrumentation overhead by ~50%.

vitalybuka committed:
[NFCI][sanitizers][metadata] Exctract create{Unlikely,Likely}BranchWeights (#89464)

We have a lot of repeated code with random constants.
Particular values are not important, the one just needs to be
bigger then another.

UR_NONTAKEN_WEIGHT is selected as it's the most common one.

clementval committed:
Revert "[flang][cuda] Use fir.cuda_deallocate for automatic deallocation (#89450)"

This reverts commit 2a632d3d9f5c70db38c617b0816deb37ef722a7b.

This has some implication on OpenACC postDeallocate action

joker-eph committed:
Revert "[clang] CTAD: Fix require-clause is not transformed." (#89476)

Reverts llvm/llvm-project#89378

Broke the windows premerge checks
https://lab.llvm.org/buildbot/#/builders/271/builds/6788

vitalybuka committed:
[NFC][hwasan] Use DEBUG_TYPE instead of "hwasan"

boomanaiden154 committed:
Reland "[python] Bump Python minimum version to 3.8 (#78828)"

This reverts commit 2dfa30d0ca6fb6991640a18e53401d82f567f8ff.

This relands commit 0a6c74e21cc6750c843310ab35b47763cddaaf32.

This patch originally caused a host of buildbot failures due to several
buildbots not having the necessary python version. That was tracked in
issue #83962, and all bots that failed in the first round have now been
updated. This is an attempt to reland the patch to see if it sticks or
if there are a number of long-running bots where this patch will cause
failures.

vitalybuka committed:
[NFC][SanCov] Pass DomTrees as const references

They are not optional.

vitalybuka committed:
[NFCI][SanCov] Remove unused default arguments

vitalybuka committed:
[NFCI][SanCov] Remove redundant copt<> defaults

vitalybuka committed:
[NFC][SanCov] Move Module and analysis callbacks into ModuleSanitizerCoverage class

Avoid passing them around.

danix800 committed:
[ASTImporter] Fix infinite recurse on return type declared inside body (#68775)

Lambda without trailing auto could have return type declared inside the
body too.

Fixes #68775

vitalybuka committed:
[NFC][SanCov] Re-format long line

cyndyishida committed:
[clang][Apple][cmake] Disable plugin support at LLVM level (#89483)

Matches up with the clang setting

owenca committed:
[clang-format] Fix a bug in annotating CastRParen before unary && (#89346)

Also fix a bug in annotating TrailingAnnotation.

Closes #61233.

aniplcc commented:
cc @RKSimon

SahilPatidar committed:
[InstCombine] fold `cond ? x : -x == 0` into `x == 0` (#85673)

Resolve #85250 
Alive2: https://alive2.llvm.org/ce/z/7DMRCy

philnik777 committed:
[libc++] Optimize char_traits a bit (#72799)

This implements two kinds of optimizations. Specifically
- `char_traits<char8_t>` uses `char` code paths; these are heavily
  optimized and the operations are equivalent
- `char16_t` and `char32_t` `find` uses `std::find` to forward to
  `wmemchr` if they have the same size

Amila-Rukshan committed:
[clang] Marshallers.h - use move semantics for 'NodeKinds' and update possible callers to use it  (#87273)

Fixes: https://github.com/llvm/llvm-project/issues/87248

Signed-off-by: amila <amila.15@cse.mrt.ac.lk>

mordante committed:
[libc++][TZDB] Renames incomplete. (#89250)

The new name uses experimental which better conveys what it means.

Smertig committed:
[Clang][Sema] Remove invalid ctor (NFC) (#82161)

`TemplateArgumentLocInventIterator` default constructor should not
exists

https://github.com/llvm/llvm-project/blob/3496927edcd0685807351ba88a7e2cfb006e1c0d/clang/lib/Sema/TreeTransform.h#L4742
because it doesn't and couldn't initialize `Self` member that is
reference:

https://github.com/llvm/llvm-project/blob/3496927edcd0685807351ba88a7e2cfb006e1c0d/clang/lib/Sema/TreeTransform.h#L4721-L4723

Instantiation of this constructor is always a compile-time error. 

Please note, that I didn't run any tests, because cannot imagine
situation where this constructor can be properly used. There are no new
tests for this fix for the same reason.

mordante committed:
[libc++] Removes a Clang 16 Windows workaround. (#88560)

karka228 committed:
[test] Avoid writing to a potentially write-protected dir (#89242)

These tests just don't check the output written to the current directory. The
current directory may be write protected e.g. in a sandboxed environment.

The Testcases that use -emit-llvm and -verify only care about stdout/stderr
and are in this patch changed to use -emit-llvm-only to avoid writing to an
output file. The verify-inlineasmbr.mir testcase that also only care about
stdout/stderr is in this patch changed to throw away the output file and just
write to /dev/null.

HerrCai0907 committed:
[mlir] fix typo in mem2reg [NFC]

HaohaiWen committed:
[InstCombine] Update BranchProbabilityAnalysis cache result (#86470)

InstCombine may invert branch condition and profile metadata. In such
case, BranchProbabilityAnalysis should also be updated.

davemgreen committed:
[AArch64] Add phase ordering tests for some basic interleaving vectorization patterns. NFC

farzonl committed:
[SPIRV][HLSL] Add mad intrinsic lowering for spirv (#89130)

- `clang/lib/CodeGen/CGBuiltin.cpp` - Add a generic mull add
implementation. Make DXIL implementation tied to target.

resolves #88944

davemgreen committed:
[LV] Fix warning about Mask being set twice. NFC

Moxinilian committed:
[nfc][github] subscribe myself to MLIR Mem2Reg PRs

vgvassilev committed:
[clang-repl] Keep the first llvm::Module empty to avoid invalid memory access. (#89031)

Clang's CodeGen is designed to work with a single llvm::Module. In many
cases
for convenience various CodeGen parts have a reference to the
llvm::Module
(TheModule or Module) which does not change when a new module is pushed.
However, the execution engine wants to take ownership of the module
which does
not map well to CodeGen's design. To work this around we clone the
module and
pass it down.

With some effort it is possible to teach CodeGen to ask the
CodeGenModule for
its current module and that would have an overall positive impact on
CodeGen
improving the encapsulation of various parts but that's not resilient to
future
regression.

This patch takes a more conservative approach and keeps the first
llvm::Module
empty intentionally and does not pass it to the Jit. That's also not
bullet
proof because we have to guarantee that CodeGen does not write on the
blueprint. However, we have inserted some assertions to catch accidental
additions to that canary module.

This change will fixes a long-standing invalid memory access reported by
valgrind when we enable the TBAA optimization passes. It also unblock
progress
on https://github.com/llvm/llvm-project/pull/84758.

kazutakahirata committed:
[memprof] Accept Schema in the constructor of RecordWriterTrait (NFC) (#89486)

The comment being deleted in this patch is not correct.  We already
construct an instance of RecordWriterTrait with Version.

This patch teaches the constructor of RecordWriterTrait to accept
Schema.  While I am at it, this patch makes Version a private
variable.

kazutakahirata committed:
[Interpreter] Fix warnings

This patch fixes:

  clang/lib/Interpreter/IncrementalParser.cpp:214:29: error: moving a
  temporary object prevents copy elision [-Werror,-Wpessimizing-move]

  clang/lib/Interpreter/IncrementalParser.cpp:232:22: error: moving a
  temporary object prevents copy elision [-Werror,-Wpessimizing-move]

vgvassilev committed:
Revert "[clang-repl] Keep the first llvm::Module empty to avoid invalid memory access. (#89031)"

This reverts commit ca090452d64e229b539a66379a3be891c4e8f3d8 and
1faf3148fdef34ce0d556ec6a4049e06cbde71b3 because it broke a darwin bot.

vgvassilev committed:
Reland "[clang-repl] Keep the first llvm::Module empty to avoid invalid memory access. (#89031)"

Original commit message: "

Clang's CodeGen is designed to work with a single llvm::Module. In many cases
for convenience various CodeGen parts have a reference to the llvm::Module
(TheModule or Module) which does not change when a new module is pushed.
However, the execution engine wants to take ownership of the module which does
not map well to CodeGen's design. To work this around we clone the module and
pass it down.

With some effort it is possible to teach CodeGen to ask the CodeGenModule for
its current module and that would have an overall positive impact on CodeGen
improving the encapsulation of various parts but that's not resilient to future
regression.

This patch takes a more conservative approach and keeps the first llvm::Module
empty intentionally and does not pass it to the Jit. That's also not bullet
proof because we have to guarantee that CodeGen does not write on the
blueprint. However, we have inserted some assertions to catch accidental
additions to that canary module.

This change will fixes a long-standing invalid memory access reported by
valgrind when we enable the TBAA optimization passes. It also unblock progress
on https://github.com/llvm/llvm-project/pull/84758.
"

This patch reverts adc4f6233df734fbe3793118ecc89d3584e0c90f and removes
the check of `named_metadata_empty` of the first llvm::Module because on darwin
clang inserts some harmless metadata which we can ignore.

boomanaiden154 committed:
Revert "Reland "[python] Bump Python minimum version to 3.8 (#78828)""

This reverts commit f2931182fc877e813974a5f53721f859bfb5b130.

This was again causing buildbot failures. #83962 has been updated with the new
failures, notifying the buildbot maintainers that they need to update their
bots.

AZero13 committed:
[X86] X86LowerTileCopy: Find dead register to use to prevent save-reload of tile register (#83628)

adk9 committed:
[mlir][python] Fix generation of Python bindings for `async` dialect (#75960)

The Python bindings generated for "async" dialect didn't include any of
the "async" dialect ops. This PR fixes issues with generation of Python
bindings for "async" dialect and adds a test case to use them.

kazutakahirata committed:
[memprof] Omit the key length for the call stack table (#89510)

The call stack table has a constant key length, so we don't need to
serialize or deserialize it for every key-data pair.  Omitting the key
length saves 0.64% of the indexed MemProf file size.

Note that it's OK to change the format because Version2 is still under
development.

topperc committed:
[RISCV] Remove extra indentation from RISCVProcessors.td.

HerrCai0907 committed:
[tidy] update check list [NFC]

tbaederr committed:
[clang][Interp][NFC] Change pointer offset to uint64

To accomodate for recent changes in array index calculations.

nikic committed:
Revert "[X86] X86LowerTileCopy: Find dead register to use to prevent save-reload of tile register (#83628)"

This reverts commit 34acbb3801515f9f41cc2d790d26072eb004ac46.

This change causes major compile-time regressions.

cor3ntin committed:
[Clang] Do not try to diagnose parameter packs in invalid pack expressions (#89257)

In a pack expression, if the id-expression is not valid, do no try to
detect whether it is a pack as that would lead to a crash trying to
print a recovery expression.

Fixes #88929

RKSimon commented on clang/utils/TableGen/MveEmitter.cpp:
```diff
@@ -660,7 +660,7 @@ class IRBuilderResult : public Result {
   std::map<unsigned, std::string> IntegerArgs;
   IRBuilderResult(StringRef CallPrefix, std::vector<Ptr> Args,
```
std::vector<Ptr> Args?

RKSimon commented on clang/utils/TableGen/MveEmitter.cpp:
```diff
@@ -660,7 +660,7 @@ class IRBuilderResult : public Result {
   std::map<unsigned, std::string> IntegerArgs;
   IRBuilderResult(StringRef CallPrefix, std::vector<Ptr> Args,
                   std::set<unsigned> AddressArgs,
```
std::set<unsigned> AddressArgs?

RKSimon commented on clang/utils/TableGen/MveEmitter.cpp:
```diff
@@ -728,7 +728,7 @@ class IRIntrinsicResult : public Result {
   std::vector<const Type *> ParamTypes;
   std::vector<Ptr> Args;
   IRIntrinsicResult(StringRef IntrinsicID, std::vector<const Type *> ParamTypes,
```
std::vector<const Type *> ParamTypes?

Commit:
[MLIR] Generalize expand_shape to take shape as explicit input (#69267)

This patch generalizes tensor.expand_shape and memref.expand_shape to
consume the output shape as a list of SSA values. This enables us to
implement generic reshape operations with dynamic shapes using
collapse_shape/expand_shape pairs.

The output_shape input to expand_shape follows the static/dynamic
representation that's also used in `tensor.extract_slice`.

Differential Revision: https://reviews.llvm.org/D140821

Co-authored-by: Ramiro Leal-Cavazos <ramiroleal050@gmail.com>

joker-eph committed:
Revert "[MLIR] Generalize expand_shape to take shape as explicit input" (#89540)

Reverts llvm/llvm-project#69267

this broke some bots.

davemgreen committed:
[AArch64] Add costs for LD3/LD4 shuffles.

Similar to #87934, this adds costs to the shuffles in a canonical LD3/LD4
pattern, which are represented in LLVM as deinterleaving-shuffle(load). This
likely has less effect at the moment than the ST3/ST4 costs as instcombine will
perform certain transforms without considering the cost.

aniplcc committed:
update some more params
